File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def recurse(self, root: Optional[TreeNode]):
        if root is None:
            return

        self.recurse(root.left)
        self.ordered_list.append(root.val)
        self.recurse(root.right)

    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.ordered_list = []
        self.recurse(root)
        return self.ordered_list[k-1]

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    levelOrder(root) {
        if (!root) return [];
        let queue = [];
        queue.push(root);
        let res = [];

        while (queue.length > 0) {
            let sublist = [];
            const levelSize = queue.length;

            for (let i = 0; i < levelSize; i++) {
                let node = queue.shift();
                sublist.push(node.val);

                if (node.left) {
                    queue.push(node.left);
                }

                if (node.right) {
                    queue.push(node.right);
                }
            }
            res.push(sublist);
        }
        return res;
    }
}
